### PR TITLE
Improve display of validation errors for scripts

### PR DESCRIPTION
**Pull request by westonruter**
## Summary

At present the list of validation errors for scripts is not helpful at all because it just says “Invalid Element: `<script>`” over and over again. You have to expand it to figure out what it is for. We can improve that by having it instead say “Invalid script” and then provide the basename of the script being included.

### Validated URL Screen

Before | After
-------|------
wordpressdev lndo site_core-dev_src_wp-admin_post php_post=2371 action=edit | wordpressdev lndo site_core-dev_src_wp-admin_post php_post=2371 action=edit amp_urls_tested=1 amp_remaining_errors=0

### Validation Error Index Screen

Before | After
-------|------
wordpressdev lndo site_core-dev_src_wp-admin_edit-tags php_taxonomy=amp_validation_error post_type=amp_validated_url (1) | wordpressdev lndo site_core-dev_src_wp-admin_edit-tags php_taxonomy=amp_validation_error post_type=amp_validated_url (2)

## Checklist

- [ ] My pull request is addressing an [open issue](https://github.com/ampproject/amp-wp/contributing/project-management.md#life-of-an-issue) (please create one otherwise).
- [ ] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/contributing/engineering.md#tests).
- [ ] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/contributing/engineering.md) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

File: includes/validation/class-amp-validated-url-post-type.php [hunk 1]
```diff
@@ -1354,12 +1354,13 @@ public static function print_admin_notice() {
...
-			$description      = json_decode( $error->description, true );
-			$sanitization     = AMP_Validation_Error_Taxonomy::get_validation_error_sanitization( $description );
-			$status_text      = AMP_Validation_Error_Taxonomy::get_status_text_with_icon( $sanitization );
-			$error_code       = isset( $description['code'] ) ? $description['code'] : 'error';
-			$error_title      = AMP_Validation_Error_Taxonomy::get_error_title_from_code( $error_code );
 			$validation_error = json_decode( $error->description, true );
+			if ( ! is_array( $validation_error ) ) {
+				$validation_error = [];
+			}
+			$sanitization     = AMP_Validation_Error_Taxonomy::get_validation_error_sanitization( $validation_error );
```
Comment (pierlon) [inline includes/validation/class-amp-validated-url-post-type.php]:
```suggestion
			$sanitization   = AMP_Validation_Error_Taxonomy::get_validation_error_sanitization( $validation_error );
```

File: includes/validation/class-amp-validated-url-post-type.php [hunk 2]
```diff
@@ -1354,12 +1354,13 @@ public static function print_admin_notice() {
...
+			$status_text      = AMP_Validation_Error_Taxonomy::get_status_text_with_icon( $sanitization );
```
Comment (pierlon) [inline includes/validation/class-amp-validated-url-post-type.php]:
```suggestion
			$status_text    = AMP_Validation_Error_Taxonomy::get_status_text_with_icon( $sanitization );
```

File: includes/validation/class-amp-validated-url-post-type.php [hunk 3]
```diff
@@ -1354,12 +1354,13 @@ public static function print_admin_notice() {
...
+			$error_title      = AMP_Validation_Error_Taxonomy::get_error_title_from_code( $validation_error );
```
Comment (pierlon) [inline includes/validation/class-amp-validated-url-post-type.php]:
```suggestion
			$error_title    = AMP_Validation_Error_Taxonomy::get_error_title_from_code( $validation_error );
```

File: includes/validation/class-amp-validated-url-post-type.php [hunk 4]
```diff
@@ -1354,12 +1354,13 @@ public static function print_admin_notice() {
...
 			$accept_all_url   = wp_nonce_url(
```
Comment (pierlon) [inline includes/validation/class-amp-validated-url-post-type.php]:
```suggestion
			$accept_all_url = wp_nonce_url(
```

File: includes/validation/class-amp-validation-error-taxonomy.php [hunk 5]
```diff
@@ -1824,11 +1824,14 @@ public static function filter_manage_custom_columns( $content, $column_name, $te
...
-						esc_html( self::get_error_title_from_code( $validation_error['code'] ) )
+						esc_html( self::get_error_title_from_code( $validation_error ) )
 					);
 				}
 
-				if ( self::INVALID_ELEMENT_CODE === $validation_error['code'] || 'duplicate_element' === $validation_error['code'] ) {
+				$is_js_script = self::is_validation_error_for_js_script_element( $validation_error );
+				if ( self::INVALID_ELEMENT_CODE === $validation_error['code'] && $is_js_script && isset( $validation_error['node_attributes']['src'] ) ) {
+					$content .= sprintf( ': <code>%s</code>', esc_html( basename( wp_parse_url( $validation_error['node_attributes']['src'], PHP_URL_PATH ) ) ) );
+				} elseif ( ( self::INVALID_ELEMENT_CODE === $validation_error['code'] || 'duplicate_element' === $validation_error['code'] ) && ! $is_js_script ) {
```
Comment (westonruter) [inline includes/validation/class-amp-validation-error-taxonomy.php]: Eventually this whole conditional block should be moved to `\AMP_Validation_Error_Taxonomy::get_error_title_from_code()`, as opposed to checking conditions twice.

**Comment by pierlon** [comment]
Also, the `tag_row_actions` filter has been deprecated (or was supposed to be from WP 3.0.0). Only one instance of it is being used in the plugin, as found here:

https://github.com/ampproject/amp-wp/blob/480a667c4e462decdb7d60debf031366a68629f5/includes/validation/class-amp-validation-error-taxonomy.php#L761

Changing that to `amp_validation_error_row_actions` should resolve that issue, along with removing the condition for verifying the taxonomy in `AMP_Validation_Error_Taxonomy::filter_tag_row_actions`:

https://github.com/ampproject/amp-wp/blob/480a667c4e462decdb7d60debf031366a68629f5/includes/validation/class-amp-validation-error-taxonomy.php#L1625

**Comment by pierlon** [comment]
Per https://github.com/ampproject/amp-wp/pull/3722#pullrequestreview-315857487, this will show:

image

I suppose it should be showing the basename of the script here as well, and nothing if its an inline script?

**Comment by westonruter** [comment]
> I suppose it should be showing the basename of the script here as well, and nothing if its an inline script?

Yes, good point! I noticed that as well. I just pushed 96267a862ee13704122ef2f1014f91912ae38e13 to fix that.

File: includes/validation/class-amp-validation-error-taxonomy.php (hunk 5 above)
Comment (westonruter) [reply includes/validation/class-amp-validation-error-taxonomy.php]: Fixed in 96267a862ee13704122ef2f1014f91912ae38e13

**Comment by westonruter** [comment]
> Also, the `tag_row_actions` filter has been deprecated in WP 5.3 (or was supposed to be from WP 3.0.0).

Good eye. I'm impressed you noticed that. Fixed in e64154dc4ed62f12a3dfb84027cd7cc7018d2740.

**Pull request merged**